src/Nacos.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Service = Ocelot.Values.Service;
+
+namespace Ocelot.Discovery.Nacos;
+
+public class Nacos : IServiceDiscoveryProvider
+{
+    private readonly INacosNamingService _client;
+    private readonly string _serviceName;
+    private readonly ILogger<Nacos> _logger;
+
+    public Nacos(string serviceName, INacosNamingService client, ILogger<Nacos> logger)
+    {
+        _client = client;
+        _serviceName = serviceName;
+        _logger = logger;
+    }
```

Comment:
After switch the type from `ILogger<Nacos>` to `IOcelotLogger`, we must inject the factory aka `IOcelotLoggerFactory` object:
```suggestion
    public Nacos(string serviceName, INacosNamingService client, IOcelotLoggerFactory factory)
    {
        _client = client;
        _serviceName = serviceName;
        _logger = factory.CreateLogger<Nacos>();
    }
```
Example in Kubernetes provider:
https://github.com/ThreeMammals/Ocelot/blob/a8ba9848844b533181c68185098defc566a84728/src/Ocelot.Provider.Kubernetes/Kube.cs#L32

**P.S.** Please note that redevelopment of unit tests is required.